pkg/cli/server/server.go

```diff
@@ -292,9 +321,13 @@ func run(app *cli.Context, cfg *cmds.Server, leaderControllers server.CustomCont
 	}
 
 	// set ServiceIPRange to the first IPv4 block, for legacy clients
+	// unless only IPv6 range given
 	serviceIPRange, err := util.GetFirst4Net(serverConfig.ControlConfig.ServiceIPRanges)
 	if err != nil {
-		return errors.Wrap(err, "cannot configure IPv4 service-cidr")
+		serviceIPRange, err = util.GetFirst6Net(serverConfig.ControlConfig.ServiceIPRanges)
+		if err != nil {
+			return errors.Wrap(err, "cannot configure IPv4/IPv6 service-cidr")
+		}
```

Comment:
Again the utils function. I am slowly thinking that this is the best recommendation I made in weeks (maybe just an ego thing) :)